lib/Doctrine/ODM/MongoDB/Aggregation/Aggregation.php

```diff
@@ -64,6 +68,45 @@ private function prepareIterator(CursorInterface $cursor): Iterator
             $cursor = new HydratingIterator($cursor, $this->dm->getUnitOfWork(), $this->classMetadata);
         }
 
-        return $this->rewindable ? new CachingIterator($cursor) : new UnrewindableIterator($cursor);
+        $iterator = $this->rewindable ? new CachingIterator($cursor) : new UnrewindableIterator($cursor);
+
+        $this->assertSearchIndexExistsForEmptyResult($iterator);
+
+        return $iterator;
+    }
+
+    /**
+     * If the server implements a server-side error for missing search indexes,
+     * this assertion can be removed.
+     *
+     * @see https://jira.mongodb.org/browse/SERVER-110974
+     * @see Configuration::setAssertSearchIndexExistsForEmptyResult()
+     *
+     * @param CachingIterator<mixed>|UnrewindableIterator<mixed> $iterator
+     */
+    private function assertSearchIndexExistsForEmptyResult(CachingIterator|UnrewindableIterator $iterator): void
+    {
+        // The iterator is always rewinded
+        if ($iterator->current()) {
+            return; // Results not empty
+        }
+
+        if (! $this->dm->getConfiguration()->assertSearchIndexExistsForEmptyResult()) {
+            return; // Feature disabled
+        }
+
+        // Search stages must be the first stage in the pipeline
+        $stage = $this->pipeline[0] ?? null;
+        if (! $stage || ! in_array(array_key_first($stage), ['$search', '$searchMeta', '$vectorSearch'], true)) {
+            return; // Not a search aggregation
+        }
+
+        $indexName = ((object) $stage[array_key_first($stage)])->index ?? 'default';
```

Comment:
We can use `array_first`, that would require the [PHP 8.5 ployfill](https://github.com/symfony/polyfill-php85/blob/65345afb52bf45a3d996295b3504ace5db1be853/Php85.php#L37-L44). Not worth it.
```suggestion
        $indexName = ((object) array_first($stage))->index ?? 'default';
```